common/lib/dependabot/update_checkers/version_filters.rb

```diff
@@ -0,0 +1,17 @@
+module Dependabot
+  module UpdateCheckers
+    module VersionFilters
+      def filter_vulnerable_versions(versions_array)
```

Comment:
This might be a personal style but I always feel nervous about mixins as a pattern. I find they often cause subtle bugs and can get hard to debug. Thoughts on exposing a class method and calling that directly instead?

e.g. `Dependabot::UpdateCheckers::VersionFilters.filter_vulnerable_versions(relevant_versions)`

Reply:
Yeah, it'd require us to pass in the `security_advisories` as well, but that'd work 👍

Reply:
Hm good point, at which point the method is pretty basic and the signature is almost the same number of characters 😀 

Mixing in all these methods also feels hairy, `security_advisories` is currently exposed in the base update checker, and the dependency is very implicit 🤔

Reply:
Yeah I think it'd be an improvement 👍

Reply:
I definitely prefer the class method/functional pattern where we just pass in the args we need over mixins but realise it's not a common pattern in ruby. It's a shame we need to pass all this global state around.

I'm not strongly adverse to getting this in as is. This might be a good refactor candidate in future if we figure out a `context` pattern ala react where we have a immutable store that includes context like update config options, ignore conditions and sec advisories.

Reply:
I think for the sake of a [dependent PR](https://github.com/dependabot/dependabot-core/pull/3910) we should keep this as is and consider options like a global `context` for down the road.

Reply:
@Nishnha agreed for sure on punting on the `context` part, but we could still implement the suggested stateless method:

```ruby
Dependabot::UpdateCheckers::VersionFilters.filter_vulnerable_versions(relevant_versions, security_advisories)
```

The downside of using a mixin like this, is that it relies on whatever class we include it in to define the `security_advisories` state. This can lead to bugs when a class doesn't, or implements it slightly differently, and it's not a patter we use a ton in the rest of the codebase.

In order to make the suggested change, we would change the method definition to:

```ruby
def self.filter_vulnerable_versions(versions_array, security_advisories)
```

Remote all the `include Dependabot::UpdateCheckers::VersionFilters`, and change the method calls to call the method directly on the module:

```diff
- filter_vulnerable_versions(relevant_versions)
+ Dependabot::UpdateCheckers::VersionFilters.filter_vulnerable_versions(relevant_versions, security_advisories)
```